_rules/image-no-text-0va7u6.md

```diff
@@ -0,0 +1,161 @@
+---
+id: 0va7u6
+name: HTML graphics contain no text
+rule_type: atomic
+description: |
+  This rule checks that images of text are not used
+accessibility_requirements:
+  wcag20:1.4.5: # Images of Text (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag20:1.4.9: # Images of Text (No Exception) (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+  - Language
+acknowledgments:
+  authors:
+    - Carlos Duarte
+  images:
+    - Letter posted from Dulverton (Somerset, England) to Bristol (England) in 1894. Released into the public domain by Adrian Pingstone.
+---
+
+## Applicability
+
+This rule applies to any HTML element that is [visible][] and not [marked as decorative][] for which one of the following is true:
+
+- the element is an `img` element where at least one of the [image sources][] in its [source set][] does not reference an SVG document; or
+- the element is an `input` element with a `type` [attribute value][] of `image` and its `src` [attribute value][] does not reference an SVG document; or
+- the element has a [computed][] [`background-image`][background-image] CSS property with at least one [`image`][css-image] that is a [url reference][url-reference] that does not reference an SVG document.
+
+## Expectation
+
+For each test target, each of the image resources referenced by its [image sources][] that do not reference an SVG document either does not contain text expressing anything in a [human language][] or it is [essential][] that the text is rendered with that specific presentation.
```

Comment:
I think this expectation may be leaving out several uses cases in relation to 1.4.5? I think at least one of the following must be true to meet the expectation:

1. image does not contain text expressing anything in a human language (already covered)
2. the particular presentation of text is essential to the information being conveyed (already covered)
3. the font, size, color, and background of the essential text can be customized (not covered)
4. CSS cannot be used to achieve the visual presentation of the essential text (not covered)

Note: assumption is that CSS is part of "technologies being used"

Reply:
Regarding points 3 and 4, if you read the applicability, the rule leaves out SVG images, which could be customised through CSS. This is made clear in the first paragraph of the background section. I don't think that, using other image types, I can use CSS to change font-family, for instance.

Reply:
I don't think this requirement is limited to customizing text through CSS and SVG only. Here are some types of images of text, that I think would pass SC 1.4.5:

- Point 3: Image of text generated by script, where the user can set the font, size, color, and background (eg. by buttons/pull-down menus provided by the script, rather than browser/CSS settings)
- Point 4: Image of a mathematical formula used because CSS can't provide the styling and MathML is not a technology being used

I think the current rule would incorrectly fail these types of images/situations?